### PR TITLE
Fix minor bugs

### PR DESCRIPTION
**Pull request by Serjobas**
## Warnings have been added

### CODE REVIEW

File: src/Print.jsx [hunk 1]
```diff
@@ -36,24 +36,36 @@ export default class Print extends React.Component {
...
-        const printOffsetLeft = elemRect && (elemRect.left - bodyRect.left);
-        const printOffsetTop = elemRect && (elemRect.top - bodyRect.top);
+        const isPrint = window.matchMedia('print').matches;
 
-        this.setState({
-          printOffsetTop,
-          printOffsetLeft,
-        });
+        if(isPrint){
+          console.log('PRINT HERE');
```
Comment (a-x-) [inline src/Print.jsx]: are we still need it
and other logs below?

File: src/Print.jsx [hunk 2]
```diff
@@ -36,24 +36,36 @@ export default class Print extends React.Component {
...
+
+          this.setState({
+            printOffsetTop,
+            printOffsetLeft,
+          });
+        }else{
+          console.log('print end');
+          this.setState({
+            printOffsetTop: 0,
+            printOffsetLeft: 0
```
Comment (a-x-) [inline src/Print.jsx]: ? R u remember about:
> unexpected initial margin-top: 0px;margin-left: 0px;
as described at errata:
https://github.com/a-x-/react-easy-print/pull/19#issuecomment-398208526

you should avoid 0 0 margins

File: src/Print.jsx [hunk 3]
```diff
@@ -65,11 +77,15 @@ export default class Print extends React.Component {
...
+    const main_ = (main || single) ? s._main : '';
     const excl_ = (exclusive || printOnly) ? s.exclusive : '';
     const isPrint = window.matchMedia('print').matches;
-    const offset_ = ((printOffsetTop || printOffsetLeft ) && main_ && isPrint) ? { marginTop: -printOffsetTop, marginLeft: -printOffsetLeft} : {marginTop: 0, marginLeft: 0};
-    return <div ref={ (el) => this.printElement = el } style={offset_} className={`${s.root} ${main_} ${excl_}`}>{children}</div>;
+    const offset_ = ((printOffsetTop || printOffsetLeft ) && main_ && isPrint) ? { marginTop: -printOffsetTop, marginLeft: -printOffsetLeft} : {};
+    const globalClassName = 'react-easy-print-print'; // using in hiden all
+    const className = `${globalClassName} ${s.root} ${main_} ${excl_}`;
+    // this.state.printOffsetLeft = 0;
+    // this.state.printOffsetTop = 0;
```
Comment (a-x-) [inline src/Print.jsx]: look like you forgot some trash?

File: src/PrintProvider.jsx [hunk 4]
```diff
@@ -43,28 +44,46 @@ export default class PrintProvider extends React.PureComponent {
     };
   }
 
-  regPrintable(key, node, isSingle) {
+  regHiddenAll() {
+    document.body.classList.add(s.hiddenAll);
+    this.hasSingle = true;
+  }
```
Comment (a-x-) [inline src/PrintProvider.jsx]: the method has an incorrect naming since it `hasSingle` aware

File: src/PrintProvider.jsx [hunk 5]
```diff
@@ -43,28 +44,46 @@ export default class PrintProvider extends React.PureComponent {
     };
   }
 
-  regPrintable(key, node, isSingle) {
+  regHiddenAll() {
+    document.body.classList.add(s.hiddenAll);
```
Comment (a-x-) [inline src/PrintProvider.jsx]: we should add some description and disclaimer:
- it's working in one moment print snapshot context
- it used to cover all of React Portals, popups and modals and etc.

File: src/Print.jsx (hunk 3 above)
Comment (Serjobas) [reply src/Print.jsx]: my bad

File: src/Print.jsx (hunk 2 above)
Comment (Serjobas) [reply src/Print.jsx]: we do avoid them ` const offset_ = ((printOffsetTop || printOffsetLeft ) && main_ && isPrint) ? { marginTop: -printOffsetTop, marginLeft: -printOffsetLeft} : {};
`

Comment (a-x-) [reply src/Print.jsx]: o-o-ok :-)

File: src/PrintProvider.jsx (hunk 4 above)
Comment (a-x-) [reply src/PrintProvider.jsx]: hm, I got 
anyways, let's consider `hideAll() {...}`

Comment (Serjobas) [reply src/PrintProvider.jsx]: What's wrong with naming ? 
This method responds to registration of class hidenAll in the body node.
hasSingle tells us than we have single ( main ) Print

**Comment by a-x-** [comment]
📦 0.6.0-beta

**Pull request merged**